relay-general/src/store/transactions.rs

```diff
@@ -275,18 +271,23 @@ impl Processor for TransactionsProcessor {
             normalize_transaction_name(&mut event.transaction)?;
         }
 
-        validate_transaction(event)?;
-
         let spans = event.spans.value_mut().get_or_insert_with(|| Vec::new());
 
         for span in spans {
-            if span.value().is_none() {
+            if let Some(span) = span.value_mut() {
+                if span.timestamp.value().is_none() {
+                    span.timestamp.set_value(event.timestamp.value().cloned());
```

Comment:
Syntactically, the event timestamp could be `None` here. Is there a validation that checks this before?

If we normalize the event timestamp somewhere else (I think we default it to the receive timestamp), we should also apply the same logic here and put the same default.

Reply:
We did a quick survey and it seems that the event timestamp is changed in these places:

* [`process_minidump`](https://github.com/getsentry/relay/blob/1edaf631c61968066bd2f905a0b8dcfa22dc7558/relay-server/src/utils/native.rs#L204), which runs as part of `create_placehodlers` _before_ light normalization, which should be fine.
* [`normalize_timestamps`](https://github.com/getsentry/relay/blob/1edaf631c61968066bd2f905a0b8dcfa22dc7558/relay-general/src/store/normalize.rs#L532): This runs _after_ transaction processor in light normalization, which would be a problem, but if I understand correctly, the clock drift processor fixes all timestamps in the event by calling [`process_child_values`](https://github.com/getsentry/relay/blob/1edaf631c61968066bd2f905a0b8dcfa22dc7558/relay-server/src/utils/native.rs#L204), so it will alter the event timestamp and the span timestamp in the same way.

Reply:
Maybe adding test to verify these findings would be a good idea.